src/util.rs

```diff
@@ -10,18 +10,57 @@ use serde::de::DeserializeOwned;
 
 use crate::errors::*;
 
-pub const USR_SHARE_PATH: &str = "/usr/share/i3status-rust";
+pub const USR_SHARE_PATH: &str = "/usr/share/";
 
 pub fn pseudo_uuid() -> usize {
     static ID: AtomicUsize = AtomicUsize::new(usize::MAX);
     ID.fetch_sub(1, Ordering::SeqCst)
 }
 
+/// Tries to find a file in the given path.
+/// This function is shorthand for the following operations:
+/// - run the path conversion handler (add suffix, etc.)
+/// - add the i3status-rust directory
+/// - add the directory `subdir` if necessary
+/// - add `file`
+/// - if the file exists, return it.
+fn lookup_file<P, F>(
+    path: Option<P>,
+    transform: F,
+    subdir: Option<&str>,
+    file: &Path,
+) -> Option<PathBuf>
+where
+    P: Into<PathBuf>,
+    F: FnOnce(&mut PathBuf),
+{
+    let mut path: PathBuf = match path {
+        Some(path) => path.into(),
+        None => return None,
+    };
+
+    transform(&mut path);
```

Comment:
Is there a reason to pass a `transofrm` function to `lookup_file()` instead of directly calling it? I mean something like
```rust
if let Some(local_share_path) = lookup_file(
        std::env::var("HOME")
            .ok()
            .map(|path| PathBuf::from(path).join(".local/share")),
        subdir,
        &file,
    ) {
        return Some(local_share_path);
    }
```
instead of
```rust
if let Some(local_share_path) = lookup_file(
        std::env::var("HOME").ok(),
        |path| path.push(".local/share/"),
        subdir,
        &file,
    ) {
        return Some(local_share_path);
    }
```

Reply:
Hm. It looks better, but I would like to remove unnecessary allocation in join function and replace it with push. I will think about replacing the callback with map.

Reply:
Both are fine, but current `lookup_file`'s API is just a little weird (to me).

Reply:
Now `lookup_file` accepts only 3 parameters (without callback and generic).